### ISSUE TITLE
SMSS Support

### ISSUE THREAD

**Issue by rdefendi** [Cynicide/RMSS-FoundryVTT#8]
Hello,

This is Robert J Defendi, the designer of Spacemaster: Privateers and its expanded rules, which I think of as SMSS. I just discovered this and loaded it on my Foundry. I'll be experimenting with it over the next bit. I'm officially putting in an enhancement request to enlarge the scope to include SMSS. If you need me to run down what would need to be added, such as the items for the SM-specific skill categories, I'd be happy to do the legwork there.

Thanks!

Bob

**Comment by Cynicide** [comment]
> Hello,
> 
> This is Robert J Defendi, the designer of Spacemaster: Privateers and its expanded rules, which I think of as SMSS. I just discovered this and loaded it on my Foundry. I'll be experimenting with it over the next bit. I'm officially putting in an enhancement request to enlarge the scope to include SMSS. If you need me to run down what would need to be added, such as the items for the SM-specific skill categories, I'd be happy to do the legwork there.
> 
> Thanks!
> 
> Bob

Hello Robert. Nice to meet you. I can't make any promises but I'm up for including as much as I can. My aim is not to include anything that would cross the line as far as ICE's intellectual property goes so at the moment I'm limiting myself to character sheet stuff and a basic dice roller.   

I've pulled my copy of Spacemaster: Privateers off the shelf and had a look, did you just want to start with making sure there are Skills and Skill Categories available for importing to the sheet? We can take it from there and see if it needs it's own character sheet.

**Comment by rdefendi** [comment]
Yeah, give me a day or two to make a character and add a weapon or two and see how basic attacks, moving maneuvers, static maneuvers, things like that work. I don't know if you've set it up so that we can add an attack chart or maneuver chart ourselves, for instance. But my INITIAL instinct is one character sheet MIGHT do, depending on how cumbersome we are able to make text labels, or if we have a quick way to swap them. At the very least it should be easy to clone the sheet over to a new sheet to handle the slightly different MP and RR structures.

But give me a few to get my feet wet. We were discussing it this weekend and discovered your work during my Saturday game. Worse case scenario, I go on vacation at the end of the week.

Also, if you want to take any of the convo private, I'm happy to, either through at DM system or discord.

**Comment by Cynicide** [comment]
Thanks, I'm fine with public. I will warn you that it's still quite early in development.  There is no combat or dice rolling in the system at the moment you can really only drag Skills, Skill Categories and Items to character sheets. Some of the basic math is taken care of as far as feeding bonuses to the various skills and skill categories.   

At the moment I'm building a fairly generic dice roller to handle the different types of rolls (High/Low Open Ended,1-100 , etc). But I haven't pushed that code public yet as it's still buggy. In addition to this I'm trying to decide how to handle the Status Record Sheet T-6.6. I didn't fancy splatting a stack of check boxes onto the sheet and since the math is included on the character sheet I was experimenting with doing the calculation automatically and providing to the Player/GM to use during a dice roll.

**Comment by rdefendi** [comment]
Well, I'm not a coder, but a buddy and I set up a paradox database to do a lot of this back in the 90s, and I made spreadsheets for character creation, so a lot of the underlying logic has been a foe I've grappled with in the past, and I'm not in a rush. Better to get it right and tackle it a feature at a time.

**Comment by rdefendi** [comment]
Ah, I looked up 6.6. Yeah, if you have the sheet doing the math, I'd say just have it auto calculate into skills. Maybe make it visible when you roll in case the player needs to put in a manual bonus to negate it for some reason, but just autocalcing the penalty was what we did on our spreadsheet, IIRC.

**Comment by rdefendi** [comment]
Would this be a good way to handle things, since we're doing all this in issue comments? Whenever you or I see something we think should be addressed, even if it's WAY in the future, we just create an issue, and we ignore it until it's time to work on it (or we have a comment)? Like I might create an issue for each of the types of maneuvers and attacks, etc, with some notes on how we might handle them, and we just let those issues sit until it's time to actually work on them? That way we keep all the separate subjects organized and if someone has a thought, they can jot it in that separate issue without cluttering an existing thread, even if we aren't working on that issue at the moment.

**Comment by Cynicide** [comment]
That sounds great. Apologies for dropping off the face of the earth, we had covid go through the household but we're all good now. I'm just about done with a few features that I wanted to get done before doing another release. In the meantime I've added all of the skill categories from the SMSS character sheet to a compendium so they can be imported into a character.



![2022-12-14 21_47_54-My Rolemaster World • Foundry Virtual Tabletop](https://user-images.githubusercontent.com/13057767/207575534-da1eed51-ee47-4f6e-8031-b8b4572e384c.png)
